cmd/query/app/static_handler.go

```diff
@@ -93,23 +91,23 @@ func NewStaticAssetsHandler(staticAssetsRoot string, options StaticAssetsHandler
 		options.Logger = zap.NewNop()
 	}
 
-	if options.NewWatcher == nil {
-		options.NewWatcher = fswatcher.NewWatcher
-	}
-
 	indexHTML, err := loadAndEnrichIndexHTML(assetsFS.Open, options)
 	if err != nil {
 		return nil, err
 	}
 
 	h := &StaticAssetsHandler{
-		options:    options,
-		assetsFS:   assetsFS,
-		newWatcher: options.NewWatcher,
+		options:  options,
+		assetsFS: assetsFS,
 	}
 
+	watcher, err := fswatcher.NewFSWatcher([]string{options.UIConfigPath}, h.reloadUIConfig, h.options.Logger)
+	if err != nil {
+		return nil, err
```

Comment:
we can add a test for this by passing invalid file name

Reply:
If UI config file name is invalid, then the error has already been returned in L96, through loadAndEnrichIndexHTML?